src/pyhf/optimize/opt_jax.py

```diff
@@ -1,70 +1,76 @@
-"""JAX Optimizer Backend."""
+"""JAX Backend Function Shim."""
 
-from .. import get_backend, default_backend
+from .. import get_backend
 from ..tensor.common import _TensorViewer
-from .autodiff import AutoDiffOptimizerMixin
 import jax
+import logging
 
+log = logging.getLogger(__name__)
 
-def _final_objective(pars, data, fixed_vals, model, objective, fixed_idx, variable_idx):
+
+def _final_objective(pars, data, fixed_values, fixed_idx, variable_idx, objective, pdf):
+    log.debug('jitting function')
     tensorlib, _ = get_backend()
-    tv = _TensorViewer([fixed_idx, variable_idx])
     pars = tensorlib.astensor(pars)
-    constrained_pars = tv.stitch([fixed_vals, pars])
-    return objective(constrained_pars, data, model)[0]
+    if len(fixed_values) > 0:
+        tv = _TensorViewer([fixed_idx, variable_idx])
+        constrained_pars = tv.stitch(
+            [tensorlib.astensor(fixed_values, dtype='float'), pars]
+        )
+    else:
+        constrained_pars = pars
+    return objective(constrained_pars, data, pdf)[0]
 
 
 _jitted_objective_and_grad = jax.jit(
-    jax.value_and_grad(_final_objective), static_argnums=(3, 4, 5, 6)
+    jax.value_and_grad(_final_objective, argnums=0), static_argnums=(3, 4, 5, 6)
 )
 
+_jitted_objective = jax.jit(_final_objective, static_argnums=(3, 4, 5, 6))
 
-class jax_optimizer(AutoDiffOptimizerMixin):
-    """JAX Optimizer Backend."""
-
-    def setup_minimize(
-        self, objective, data, pdf, init_pars, par_bounds, fixed_vals=None
-    ):
-        """
-        Prepare Minimization for AutoDiff-Optimizer.
-
-        Args:
-            objective: objective function
-            data: observed data
-            pdf: model
-            init_pars: initial parameters
-            par_bounds: parameter boundaries
-            fixed_vals: fixed parameter values
-
-        """
 
-        tensorlib, _ = get_backend()
-        all_idx = default_backend.astensor(range(pdf.config.npars), dtype='int')
-        all_init = default_backend.astensor(init_pars)
+def wrap_objective(objective, data, pdf, stitch_pars, do_grad=False, jit_pieces=None):
```

Comment:
`stitch_pars` is not used anymore

Reply:
Nope, but it's used in all other shims and we keep the same signature. I can `_` it instead.

Reply:
hm maybe we rather put 

```
def stitch_pars(fixed_values,fixed_idx,variable_idx)
    if len(fixed_values) > 0:
        tv = _TensorViewer([fixed_idx, variable_idx])
        constrained_pars = tv.stitch(
            [tensorlib.astensor(fixed_values, dtype='float'), pars]
        )
    else:
        constrained_pars = pars
```

into some common.py and all shims use this?

Reply:
`_internal_postprocess` also uses `stitch_pars` so it would have to be refactored too.

Reply:
ok